File: python/AbstractFactory/Base/ioclient.py
"""
ioclient.py
"""

from input import *
from output import *
from iofactory import *

def main():
    factory = FileIOFactory()
    input = factory.create_input()
    output = factory.create_output()
    input.read()
    output.write()

    factory = ConsoleIOFactory()
    input = factory.create_input()
    output = factory.create_output()
    input.read()
    output.write()

    factory = NetIOFactory()
    input = factory.create_input()
    output = factory.create_output()
    input.read()
    output.write()

if __name__ == '__main__':
    main()

File: cc/AbstractFactory/Base/ioclient.cc
#include "input.h"
#include "output.h"
#include "iofactory.h"

int main(int argc, char* argv[]) {
    IOFactory* factory = new FileIOFactory();
    Input* input = factory->create_input();
    Output* output = factory->create_output();
    input->read();
    output->write();
    delete factory;
    delete input;
    delete output;

    factory = new ConsoleIOFactory();
    input = factory->create_input();
    output = factory->create_output();
    input->read();
    output->write();
    delete factory;
    delete input;
    delete output;

    factory = new NetIOFactory();
    input = factory->create_input();
    output = factory->create_output();
    input->read();
    output->write();
    delete factory;
    delete input;
    delete output;

    return 0;
}

